### ISSUE TITLE
\OC\Files\Filesystem::initMountPoints() misuses OCP\Files\Config\IMountPointProviderCollection

### ISSUE THREAD

**Issue by dagan** [owncloud/core#15997]
```php
// Chance to mount for other storages
if($userObject) {
	$mountConfigManager = \OC::$server->getMountProviderCollection();
	$mounts = $mountConfigManager->getMountsForUser($userObject);
	array_walk($mounts, array(self::$mounts, 'addMount'));
}
```
https://github.com/owncloud/core/blob/master/lib/private/files/filesystem.php#L414-L419

```OC\Files\Filesystem::initMountPoints()``` retrieves the ```OCP\Files\Config\IMountProviderCollection``` object and calls ```getMountsForUser()```, which returns an array of ```OCP\Files\Mount\IMountPoint``` objects. The ```IMountPoint``` objects are then passed to ```OC\Files\Mount\Manager::addMount()```, which expects an ```OC\Files\Mount\MountPoint``` object.  ```OC\Files\Mount\MountPoint``` implements ```OCP\Files\Mount\IMountPoint``` but not all ```IMountPoint``` objects are instances of ```OC\Files\Mount\MountPoint```.

**Comment by LukasReschke** [comment]
cc @icewind1991 @DeepDiver1975 @MorrisJobke

**Comment by PVince81** [comment]
@icewind1991 didn't you fix this recently ? (regarding the mismatch interface / class)

**Comment by Xenopathic** [comment]
Indeed, fixed in master as of 57945dbc4b0144bdebeda5cf15a168556118e9e3

Perhaps we want to backport that commit?

**Comment by DeepDiver1975** [comment]
> Perhaps we want to backport that commit?

I see no reason for a backport
